src/main/java/gregtech/api/metatileentity/WorkableTieredMetaTileEntity.java

```diff
@@ -125,16 +125,15 @@ protected boolean canInputFluid(FluidStack inputFluid) {
                     matchingRecipes = new HashSet<>(recipeMap.getRecipesForFluid(fluidInTank));
                 } else {
                     //else, remove recipes that don't contain fluid in this tank from list
-                    matchingRecipes.removeIf(recipe -> !recipe.hasInputFluid(fluidInTank));
+                    matchingRecipes.retainAll(recipeMap.getRecipesForFluid(fluidInTank));
                 }
             }
         }
         if (matchingRecipes == null) {
             //if all tanks are empty, generally fluid can be inserted if there are recipes for it
             return !recipeMap.getRecipesForFluid(inputFluid).isEmpty();
         } else {
-            //otherwise, we can insert fluid only if one of recipes accept it as input
-            return matchingRecipes.stream().anyMatch(recipe -> recipe.hasInputFluid(inputFluid));
+            return true;
```

Comment:
What was the meaning behind this change, as it changes the logic of adding fluids from how things currently are and I don't see how it related to the non-consumable inputs.

Currently, you cannot place a fluid into a machine if there is already a fluid in there and the fluid you are trying to place into the machine does not have a recipe with the existing fluid, which is quite a nice feature in my opinion.